### PR TITLE
Issue #764: Improve comment settings on both node form and node type for...

### LINKED ISSUE #764: [UX] Improve comment settings on both node form and node type form
Not all users get the difference between the two. How about some help text there. This is from [Administering a content type's comment settings](https://www.drupal.org/node/777818):

>- Hidden: No comments are allowed, and past comments are hidden.
>- Closed: No comments are allowed, but any past comments remain visible.
>- Open: Any future content of this type is open to new comments.

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/764

### CODE REVIEW

**Comment by jenlampton** [comment]
PR also includes the renaming of comment-node-form.js to comment.js since I discovered that it is needed on both the node form and the content type form.

**Comment by quicksketch** [comment]
For data consistency, this needs an update hook to change all existing node types that have the default comment setting of "hidden" to be set to "closed".

I'm also not super-keen on renaming the JS file, in the event that later we need a comment.js that is used for front-end (e.g. AJAX-submit of comments?) But I don't feel strongly about it.

**Comment by jenlampton** [comment]
Agree about the back-end vs front-end comment js.  Renamed comment.admin.js to be consistent with system.admin.js that's only added on the back end.

Also, latest PR contains a update hook!

**Comment by klonos** [comment]
> Renamed comment.admin.js to be consistent with system.admin.js ...

I still see it as `comment.js`

**Comment by jenlampton** [comment]
It helps if you remember to commit your changes...  pushed again

**Comment by quicksketch** [comment]
Could you fix the typo in the commit message? "commtent settings"

**Comment by jenlampton** [comment]
pushed

File: core/modules/comment/comment.install [hunk 1]
```diff
@@ -352,6 +351,20 @@ function comment_update_1002() {
 }
 
 /**
+ * Node type comment settings should be closed instead of hidden by default.
+ */
+function comment_update_1003() {
+  foreach (node_type_get_names() as $type => $name) {
```
Comment (quicksketch) [inline core/modules/comment/comment.install]: For the best compatibility and lowest likelihood of getting incorrectly cached data, could we switch this to `foreach (config_get_names_with_prefix('node.type.') as $config_name)`?

File: core/modules/comment/comment.module [hunk 2]
```diff
@@ -1239,28 +1238,41 @@ function comment_form_node_form_alter(&$form, $form_state) {
...
+ * Implements hook_node_presave().
+ */
+function comment_node_presave(Node $node) {
+  if ($node->comment == COMMENT_NODE_CLOSED && property_exists($node, 'comment_hidden') && $node->comment_hidden) {
+    $node->comment = COMMENT_NODE_HIDDEN;
+    unset($node->comment_hidden);
+  }
+}
+
+
```
Comment (quicksketch) [inline core/modules/comment/comment.module]: Extra line here could probably be removed.

File: core/modules/comment/comment.install (hunk 1 above)
Comment (jenlampton) [inline core/modules/comment/comment.install]: like this?

```
function comment_update_1003() {
  foreach (config_get_names_with_prefix('node.type.') as $config_name)
    $config = config($config_name);
    $comment_default = $config->get('settings.comment_default');
    if ($comment_default == COMMENT_NODE_HIDDEN) {
      $config->set('settings.comment_default', COMMENT_NODE_CLOSED);
      $config->save();
    }
  }
}
```

**Comment by quicksketch** [comment]
This looks great! The only thing lacking is tests. I've filed a PR that adds tests over at https://github.com/backdrop/backdrop/pull/1149. All other code left exactly the same.

**Pull request closed without merging**